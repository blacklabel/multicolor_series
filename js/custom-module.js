(function (factory) {
  if (typeof module === 'object' && module.exports) {
		module.exports = factory;
	} else {
		factory(Highcharts);
	}
}(function (Highcharts) {
  const _modules = Highcharts ? Highcharts._modules : {};

  function _registerModule(obj, path, args, fn) {
      if (!obj.hasOwnProperty(path)) {
          obj[path] = fn.apply(null, args);

          if (typeof CustomEvent === 'function') {
              window.dispatchEvent(new CustomEvent(
                  'HighchartsModuleLoaded',
                  { detail: { path: path, module: obj[path] } }
              ));
          }
      }
  }

  _registerModule(
    _modules,
    'Extensions/ColoredLineSeries.js',
    [_modules['Core/Series/SeriesRegistry.js']],
    function (SeriesRegistry) {

      // ----- Module code ----- //
      const { line: LineSeries } = SeriesRegistry.seriesTypes;

      class ColoredLineSeries extends LineSeries {
        drawGraph() {  
          // Call the base method
          super.drawGraph.apply(this);

          console.log('Inside coloredline drawGraph method!');
        }
      }

      SeriesRegistry.registerSeriesType('coloredline', ColoredLineSeries);
      // ----- Module code ----- //

    }
  );
}));
