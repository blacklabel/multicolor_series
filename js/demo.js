Highcharts.chart('showcase-demo', {
  chart: {
    spacing: [30, 30, 20, 30]
  },
  title: {
    text: 'Multicolor Series'
  },
  subtitle: {
    useHTML: true,
    align: 'right',
    floating: true,
    x: 5,
    y: 5,
    text: `
			<a
				href="https://blacklabel.net/highcharts"
				title="blacklabel.net"
			>
				<img
					src="https://blacklabel.net/wp-content/uploads/2023/09/blacklabel_logo_white.png"
					title="BlackLabel"
					alt="BlackLabel"
					style="height: 29px; filter: brightness(0%);"
				>
			</a>
		`
  },
  credits: {
    position: {
      x: -35,
      y: -30
    }
  },
  plotOptions: {
    series: {
      marker: {
        enabled: false
      }
    }
  },
  yAxis: {
    title: {
      text: ''
    },
    labels: {
      enabled: false
    }
  },
  series: [
    {
      type: 'coloredline',
      name: 'Line series',
      data: [
        {
          y: 12,
          segmentColor: '#7cb5ec'
        },
        {
          y: 15,
          segmentColor: '#7cb5ec'
        },
        {
          y: 18,
          segmentColor: '#90ed7d'
        },
        {
          y: 23,
          segmentColor: '#90ed7d'
        },
        {
          y: 29,
          segmentColor: '#90ed7d'
        },
        {
          y: 24,
          segmentColor: '#f7a35c'
        },
        {
          y: 11,
          segmentColor: '#f7a35c'
        },
        {
          y: 8,
          segmentColor: '#8085e9'
        },
        {
          y: 9,
          segmentColor: '#8085e9'
        }
      ]
    },
    {
      type: 'coloredarea',
      name: 'Area series',
      pointStart: 9,
      data: [
        {
          y: 10,
          segmentColor: '#f15c80'
        },
        {
          y: 13,
          segmentColor: '#e4d354'
        },
        {
          y: 19,
          segmentColor: '#e4d354'
        },
        {
          y: 22,
          segmentColor: '#2b908f'
        },
        {
          y: 20,
          segmentColor: '#91e8e1'
        },
        {
          y: 18,
          segmentColor: '#91e8e1'
        },
        {
          y: 17,
          segmentColor: '#91e8e1'
        },
        {
          y: 14,
          segmentColor: '#f45b5b'
        },
        {
          y: 15,
          segmentColor: '#f45b5b'
        }
      ]
    }
  ]
});

Highcharts.chart('coloredline', {
  title: {
    text: 'Colored line series'
  },
  subtitle: {
    useHTML: true,
    align: 'right',
    floating: true,
    x: 5,
    y: 5,
    text: `
			<a
				href="https://blacklabel.net/highcharts"
				title="blacklabel.net"
			>
				<img
					src="https://blacklabel.net/wp-content/uploads/2023/09/blacklabel_logo_white.png"
					title="BlackLabel"
					alt="BlackLabel"
					style="height: 29px; filter: brightness(0%);"
				>
			</a>
		`
  },
  plotOptions: {
    series: {
      marker: {
        enabled: false
      }
    }
  },
  series: [
    {
      type: 'coloredline',
      name: 'Line series',
      data: [
        {
          y: 12,
          segmentColor: '#7cb5ec'
        },
        {
          y: 15,
          segmentColor: '#7cb5ec'
        },
        {
          y: 18,
          segmentColor: '#90ed7d'
        },
        {
          y: 23,
          segmentColor: '#90ed7d'
        },
        {
          y: 29,
          segmentColor: '#90ed7d'
        },
        {
          y: 24,
          segmentColor: '#f7a35c'
        },
        {
          y: 11,
          segmentColor: '#f7a35c'
        },
        {
          y: 8,
          segmentColor: '#8085e9'
        },
        {
          y: 9,
          segmentColor: '#8085e9'
        }
      ]
    }
  ]
});

Highcharts.chart('coloredarea', {
  title: {
    text: 'Colored area series'
  },
  subtitle: {
    useHTML: true,
    align: 'right',
    floating: true,
    x: 5,
    y: 5,
    text: `
			<a
				href="https://blacklabel.net/highcharts"
				title="blacklabel.net"
			>
				<img
					src="https://blacklabel.net/wp-content/uploads/2023/09/blacklabel_logo_white.png"
					title="BlackLabel"
					alt="BlackLabel"
					style="height: 29px; filter: brightness(0%);"
				>
			</a>
		`
  },
  plotOptions: {
    series: {
      marker: {
        enabled: false
      }
    }
  },
  series: [
    {
      type: 'coloredarea',
      name: 'Area series',
      data: [
        {
          y: 10,
          segmentColor: '#f15c80'
        },
        {
          y: 13,
          segmentColor: '#e4d354'
        },
        {
          y: 19,
          segmentColor: '#e4d354'
        },
        {
          y: 22,
          segmentColor: '#2b908f'
        },
        {
          y: 20,
          segmentColor: '#91e8e1'
        },
        {
          y: 18,
          segmentColor: '#91e8e1'
        },
        {
          y: 17,
          segmentColor: '#91e8e1'
        },
        {
          y: 14,
          segmentColor: '#f45b5b'
        },
        {
          y: 15,
          segmentColor: '#f45b5b'
        }
      ]
    }
  ]
});
