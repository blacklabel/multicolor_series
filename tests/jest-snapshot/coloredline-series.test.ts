import Highcharts from 'highcharts';
import multicolorModule from '../../js/multicolor_series.js';
import { ChartSeriesColored } from '../../types.js';
import { generateFormattedSegments, generateFormattedSeries } from './helper';

multicolorModule(Highcharts);

describe("The data structure for coloredline series should be appropriate.", () => {
    const containerElement = document.createElement("div");
    const chart = Highcharts.chart(containerElement, {
        accessibility: {
          enabled: false
        },

        series: [
            {
                type: "coloredline",
                data: [
                    {
                        y: 10,
                        segmentColor: "red"
                    },
                    {
                        y: 20,
                        segmentColor: "blue"
                    },
                    {
                        y: 30,
                        segmentColor: "yellow"
                    },
                    {
                        y: 40,
                        segmentColor: "green"
                    },
                    {
                        y: 50,
                        segmentColor: "pink"
                    }
                ]
            }
        ]
    }) as ChartSeriesColored;

    const series = chart.series[0];

    if (series) {
        const data = series.data;

        test('The series data should be appropriate.', () => {
            const formattedSeries = generateFormattedSeries(series, data);
            expect(formattedSeries).toMatchSnapshot();
        });

        test('The segments data should be appropriate.', () => {
            const formattedSegments = generateFormattedSegments(series.segments, data);
            expect(formattedSegments).toMatchSnapshot();
        });
    }
});
