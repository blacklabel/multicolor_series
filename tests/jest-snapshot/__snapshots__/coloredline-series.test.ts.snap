// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The data structure for coloredline series should be appropriate. The segments data should be appropriate. 1`] = `
[
  {
    "color": "red",
    "points": [
      {
        "index": 0,
        "segmentColor": "red",
        "x": 0,
        "y": 10,
      },
      {
        "index": 1,
        "segmentColor": "blue",
        "x": 1,
        "y": 20,
      },
    ],
  },
  {
    "color": "blue",
    "points": [
      {
        "index": 1,
        "segmentColor": "blue",
        "x": 1,
        "y": 20,
      },
      {
        "index": 2,
        "segmentColor": "yellow",
        "x": 2,
        "y": 30,
      },
    ],
  },
  {
    "color": "yellow",
    "points": [
      {
        "index": 2,
        "segmentColor": "yellow",
        "x": 2,
        "y": 30,
      },
      {
        "index": 3,
        "segmentColor": "green",
        "x": 3,
        "y": 40,
      },
    ],
  },
  {
    "color": "green",
    "points": [
      {
        "index": 3,
        "segmentColor": "green",
        "x": 3,
        "y": 40,
      },
      {
        "index": 4,
        "segmentColor": "pink",
        "x": 4,
        "y": 50,
      },
    ],
  },
]
`;

exports[`The data structure for coloredline series should be appropriate. The series data should be appropriate. 1`] = `
{
  "data": [
    {
      "segmentColor": "red",
      "x": 0,
      "y": 10,
    },
    {
      "segmentColor": "blue",
      "x": 1,
      "y": 20,
    },
    {
      "segmentColor": "yellow",
      "x": 2,
      "y": 30,
    },
    {
      "segmentColor": "green",
      "x": 3,
      "y": 40,
    },
    {
      "segmentColor": "pink",
      "x": 4,
      "y": 50,
    },
  ],
  "seriesType": "line",
}
`;
